pinot-spi/src/main/java/org/apache/pinot/spi/config/table/ingestion/ComplexTypeHandlingConfig.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table.ingestion;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.pinot.spi.config.BaseJsonConfig;
+
+
+/**
+ * Config related to handling complex type
+ */
+public class ComplexTypeHandlingConfig extends BaseJsonConfig {
+  public enum Mode {
+    NONE, FLATTEN_MAP
+  }
+
+  @JsonPropertyDescription("The complex-type handling mode")
+  private final Mode _mode;
+
+  @JsonPropertyDescription("The collections to unnest")
+  private final List<String> _unnestConfig;
+
+  @JsonCreator
+  public ComplexTypeHandlingConfig(@JsonProperty(value = "mode") @Nullable Mode mode,
```

Comment:
when can mode be null? If mode is null, the config wouldn't even be in the table config rt?

Reply:
Good point. This shall be required field